From: Valentine Fatiev <valentinef@nvidia.com>
Subject: [PATCH] BACKPORT:
 drivers/net/ethernet/mellanox/mlx5/core/en/tc/act/ct.c

Change-Id: If643d7b0c6c6cc04e7366bcee61fb3c716699736
---
 drivers/net/ethernet/mellanox/mlx5/core/en/tc/act/ct.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/net/ethernet/mellanox/mlx5/core/en/tc/act/ct.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en/tc/act/ct.c
@@ -52,8 +52,10 @@ tc_act_is_multi_table_act_ct(struct mlx5
 			     const struct flow_action_entry *act,
 			     struct mlx5_flow_attr *attr)
 {
+#ifdef HAVE_FLOW_ACTION_CT
 	if (act->ct.action & TCA_CT_ACT_CLEAR)
 		return false;
+#endif
 
 	return true;
 }
@@ -61,10 +63,14 @@ tc_act_is_multi_table_act_ct(struct mlx5
 static bool
 tc_act_is_missable_ct(const struct flow_action_entry *act)
 {
+#ifdef HAVE_FLOW_ACTION_ENTRY_MISS_COOKIE
 	if (act->ct.action & TCA_CT_ACT_CLEAR)
 		return false;
 
 	return true;
+#else
+	return false;
+#endif
 }
 
 struct mlx5e_tc_act mlx5e_tc_act_ct = {
